bedrock/mozorg/templates/mozorg/about/manifesto-details.html

```diff
@@ -84,7 +84,7 @@ <h3>{{ ftl('manifesto-details-invitation') }}</h3>
   <div class="side-reference">
     <h4 class="side-reference-title">{{ ftl('about-shared-our-products') }}</h4>
     <p>{{ ftl('about-shared-software-innovations') }}</p>
-    <a class="more" href="{{ url('firefox.new') }}">{{ ftl('ui-learn-more') }}</a>
+    <a class="more" href="https://www.firefox.com/?redirect_source=mozilla-org">{{ ftl('ui-learn-more') }}</a>
```

Comment:
hmh we might want folks to stay on the site, so perhaps?
```suggestion
    <a class="more" href="{{ url('products.landing') }}">{{ ftl('ui-learn-more') }}</a>
```